### ISSUE TITLE
Threads

### ISSUE THREAD

**Issue by SzotsGabor** [RyanLauQF/BLANK-Chess#2]
Hi Ryan,

Frankly, I do not like the idea of reserving a separate thread for polling input. Without going into details, see the issue with Walleye at https://github.com/MitchelPaulin/Walleye/issues/12#issuecomment-953155832

**Comment by RyanLauQF** [comment]
I see... I separated the thread for search for the same reason that Walleye did. I will see what I can do to address this issue! Thanks for bringing it up!

**Comment by SzotsGabor** [comment]
Anyway, I started a tournament because it seems to me that CPU usage does not often go above 30 % so maybe that can be tolerated. Once I observed 42 %. This is on a 4-core computer so 25 % is one core.

**Comment by RyanLauQF** [comment]
Alright! I have removed the separate thread for search and instead implemented the querying of UCI commands into the main search thread itself by polling for user inputs periodically during the search. I will include the changes in the subsequent release.

**Comment by SzotsGabor** [comment]
Fine, thanks.

FYI, I have started a thread at CCC with this topic, maybe it is of interest to you what the programmers had to say.
http://talkchess.com/forum3/viewtopic.php?f=2&t=78548&sid=10f4703afb80e5eeaa94fd52ad260cfd

**Comment by SzotsGabor** [comment]
I regret to tell you that I had to stop the tournament because my computer lost its responsiveness. I'm going to wait for the next release.

**Comment by RyanLauQF** [comment]
Hi Gabor, 

I have just released BLANK v1.2.0 with a bunch of changes made. Do let me know if it fixes the cpu usage issue. Thank you!

**Comment by SzotsGabor** [comment]
Hi Ryan,

A quick test showes that CPU usage is fluctuating slightly around 25 %, I could observe values between 23 and 28 (roughly, it changes so quickly that it is hard to follow it by the eye).

**Comment by RyanLauQF** [comment]
That's good to hear. I will close this thread in a day or two if there are no issues with the latest version.

**Comment by SzotsGabor** [comment]
Could be that I start a new tournament in a couple of hours. Do you think the strength is about the same as that of v1.1.0?

**Comment by RyanLauQF** [comment]
There should be a noticeable increase in strength of the engine. The static evaluation of the engine and transposition table has been changed and I have also fixed a couple of bugs in the engine's search along the way.

**Comment by RyanLauQF** [comment]
I did some small scale testing against v1.1.0 under Time control of 1min / 1s for 55 games.
v1.2.0 seems to be stronger than v1.1.0, though the sample size may be small.

Results:
>BLANK-v1.2.0 - BLANK-v1.1.0 : 
44.5/55
39-5-11 (W-L-D)

**Comment by SzotsGabor** [comment]
Yes, seems stronger. I am currently running it against Walleye, Sabrina, Quokka, Hactar, miniMardi, Suff, Cefap, Goyaz, Jacky, Blunder, Joanna, Nanook, miniChessAI, Virutor, Lynx and Ozwald and it has 20 out of 32 right now.
If it turns out that the selected opponents are too weak I'm going to select stronger ones next time.
I am too late for this week's CCRL update so you are going to see my results only next Sunday.
